project/src/main/java/com/google/sps/servlets/EventRankerServlet.java

```diff
@@ -18,26 +19,26 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.javatuples.Pair;
 
 @WebServlet("/event-ranker")
 public class EventRankerServlet extends HttpServlet {
   private static String EVENTS_KEY = "events";
- 
+
   /** Returns events in order of relevance */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> eventIds = new ArrayList<>();
-    String[] eventIdsParameter = (request.getParameter(EVENTS_KEY).split(","));
-    for (String event : eventIdsParameter) {
-        eventIds.add(event.replace("\"", "").replace("[", "").replace("]", ""));
-    }
     String email = UserServiceFactory.getUserService().getCurrentUser().getEmail();
     Optional<User> userOptional = SpannerTasks.shallowReadUserFromEmail(email);
     User user;
     if (userOptional.isPresent()) {
+      ArrayList<String> eventIds = new ArrayList<>();
+      String[] eventIdsParameter = (request.getParameter(EVENTS_KEY).split(","));
+      for (String event : eventIdsParameter) {
+          eventIds.add(event.replace("\"", "").replace("[", "").replace("]", ""));
+      }
       user = userOptional.get();
-      Set<Event> relevantEvents = new HashSet<Event>(SpannerTasks.getEventsFromIds(eventIds));
-      List<Event> rankedRelevantEvents = EventRanker.rankEvents(user, relevantEvents);
+      List<Event> rankedRelevantEvents = EventRanker.rankEvents(user, SpannerTasks.getEventsFromIds(eventIds));
```

Comment:
can inline this if you choose.

Reply:
I think this is in-lined as far as it can be, since I'm directly passing in the results from SpannerTasks